tests/libnet/ping.go

```diff
@@ -32,10 +32,10 @@ import (
 
 // PingFromVMConsole performs a ping through the provided VMI console.
 // Optional arguments for the ping command may be provided, overwirting the default ones.
-// (default ping options: "-c 1, -w 5")
-// Note: The maximum overall command timeout is 10 seconds.
+// (default ping options: "-c 5, -w 10")
+// Note: The maximum overall command timeout is 20 seconds.
 func PingFromVMConsole(vmi *v1.VirtualMachineInstance, ipAddr string, args ...string) error {
-	const maxCommandTimeout = 10 * time.Second
+	const maxCommandTimeout = 20 * time.Second
```

Comment:
can derive from args, in case there is "-w" one,
so it won't affect all the calls of PingFromVMConsole

Reply:
I think it would make sense to calculate this from parameters `c*w`.

Reply:
`ping` man for the `-c` flag states: `Stop after sending count ECHO_REQUEST packets. With deadline option, ping waits for count ECHO_REPLY packets, until the timeout expires.` , therefore, this will not help if one of the echo requests are not answered at all.

Reply:
In case of one of the echo is not answered, it will not reach `count` ECHO_REPLY, and will expire on `deadline`
Which is good, we want 5 packets that all are answered, unless we want to reduce it to 1 (currently its 5 on master).

do this `ping 8.8.8.8 -c 5 -w 1` locally and see 1 packet received and it then stops
and return 1 which what happend in the failure i think.

More info about `-w`
```
       -w deadline
              Specify a timeout, in seconds, before ping exits regardless of how many packets have been sent or received. In this case ping does not stop after  count  packet
              are sent, it waits either for deadline expire or until count probes are answered or for some error notification from network.

       -W timeout
              Time to wait for a response, in seconds. The option affects only timeout in absence of any responses, otherwise ping waits for two RTTs.
```

Reply:
Ah I see, thanks.

Reply:
@oshoval , I think you missed the main point. If the first echo request is lost (the request itself or the answer), even if you wait 1 year the ping eventually will return 1.

Reply:
c is 1, but it still send 5 packets, if suddenly one will get answer it will success 
```
$ time ping 8.8.8.18 -c 1 -w 5
PING 8.8.8.18 (8.8.8.18) 56(84) bytes of data.

--- 8.8.8.18 ping statistics ---
5 packets transmitted, 0 received, 100% packet loss, time 3999ms

real	0m5.010s
user	0m0.001s
sys	0m0.010s
```

I just tried that, rebooted my remote machine, pinged -c 1 -w 200
```
$ ping hades05.lab.eng.tlv2.redhat.com -c 1 -w 200
PING hades05.lab.eng.tlv2.redhat.com (10.46.8.59) 56(84) bytes of data.
64 bytes from hades05.lab.eng.tlv2.redhat.com (10.46.8.59): icmp_seq=173 ttl=57 time=18.0 ms

--- hades05.lab.eng.tlv2.redhat.com ping statistics ---
173 packets transmitted, 1 received, 99% packet loss, time 172002ms
rtt min/avg/max/mdev = 18.015/18.015/18.015/0.000 ms
$ echo $?
0
```

It works perfect, it just need c fine, and it tries much better than that (see 173)

Reply:
I disagree and per the logs there was only 1 response out of 5 required.
If any of the responses are missing, in 10, 20 or 10000 seconds, it will fail.
It us unlikely that the round-trip tool more than a few msec, but even if it took 1sec each, it would still pass.

As stated before, in case a packet is lost, which I suspect this is what happened here, then increasing the deadline will not help.

Reply:
I just gave you a real life example, i rebooted a machine, and then started ping, once its back, it need only `c` packets which are fine, and it returned success.

You mixed, the first example ping non valid address (8.8.8.18 , not 8.8.8.8), and all are failed
the 2nd example need only 1 packet (-c 1) and it success. "1 received"